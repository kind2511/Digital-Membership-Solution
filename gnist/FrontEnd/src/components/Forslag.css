.forslag-container {
  background-color: #f2f2f2;
  border: 1px solid #ccc;
  border-radius: 8px;
  padding: 20px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  max-height: 400px;
  overflow-y: auto;
}
.forslag-container .section-title {
  color: #333;
  font-size: 20px;
  margin-bottom: 10px;
}
.suggestion-item {
  border-bottom: 1px solid #e1e1e1;
  padding-bottom: 10px;
  margin-bottom: 10px;
}
.suggestion-item:last-child {
  border-bottom: none;
}

.delete-suggestion-button {
  background-color: red;
  color: white;
  border: none;
  padding: 8px 16px;
  cursor: pointer;
  border-radius: 4px;
}

.delete-suggestion-button:hover {
  background-color: #d32f2f;
}

/* Success message */
.forslag-success-message {
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(40, 167, 69, 0.9);
  color: white;
  text-align: center;
  padding: 10px;
  font-weight: bold;
  z-index: 1500;
}
