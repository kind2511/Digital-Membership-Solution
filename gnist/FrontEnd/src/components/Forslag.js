import React, { useState, useEffect } from 'react';
import './Forslag.css';

function Forslag() {
  const [suggestions, setSuggestions] = useState([]);
  const [showSuccessMessage, setShowSuccessMessage] = useState(false);

  useEffect(() => {
    fetchSuggestions();
  }, []);

  const fetchSuggestions = async () => {
    try {
      const response = await fetch('http://127.0.0.1:8000/digital_medlemsordning/get_all_suggestions/');
      if (!response.ok) {
        throw new Error('Network response was not ok');
      }
      const data = await response.json();
      setSuggestions(data);
    } catch (error) {
      console.error('Fetch error:', error);
    }
  };

  const handleDeleteSuggestion = async (suggestionID) => {
    try {
      const response = await fetch(`http://127.0.0.1:8000/digital_medlemsordning/delete_suggestion/${suggestionID}/`, {
        method: 'DELETE',
      });
      if (!response.ok) {
        throw new Error('Failed to delete suggestion');
      }
      setShowSuccessMessage(true);
      setTimeout(() => setShowSuccessMessage(false), 3000);
      await fetchSuggestions();
    } catch (error) {
      console.error('Error deleting suggestion:', error);
    }
  };

  return (
    <div className="forslag-container">
      <h2 className="section-title">Forslag</h2>
      <div className="section-content">
        {suggestions.map((suggestion) => (
          <div key={suggestion.suggestionID} className="suggestion-item">
            <h3>{suggestion.title}</h3>
            <p>{suggestion.description}</p>
            <button
              className="delete-suggestion-button"
              onClick={() => handleDeleteSuggestion(suggestion.suggestionID)}
            >
              Slett
            </button>
          </div>
        ))}
      </div>
      {showSuccessMessage && (
        <div className="delete-suggestion-success-message ">Forslaget ble slettet.</div>
      )}
    </div>
  );
}

export default Forslag;
