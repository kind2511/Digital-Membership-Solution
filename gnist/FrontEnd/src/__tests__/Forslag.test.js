import React from 'react';
import renderer from 'react-test-renderer';
import Forslag from '../components/Forslag';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);
const suggestionsMock = [
  {
    suggestionID: 1,
    title: 'Suggestion 1',
    description: 'Description for suggestion 1'
  },
  {
    suggestionID: 2,
    title: 'Suggestion 2',
    description: 'Description for suggestion 2'
  }
];
mock.onGet('http://127.0.0.1:8000/digital_medlemsordning/get_all_suggestions/').reply(200, suggestionsMock);

test('Forslag component snapshot', () => {
  const component = renderer.create(<Forslag />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
