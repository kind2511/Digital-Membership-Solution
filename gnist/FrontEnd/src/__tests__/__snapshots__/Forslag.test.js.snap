// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forslag component snapshot 1`] = `
<div
  className="forslag-container"
>
  <h2
    className="section-title"
  >
    Forslag
  </h2>
  <div
    className="section-content"
  />
</div>
`;
